R/retrieveData-KNMI-OpenAPI.R

```diff
@@ -0,0 +1,218 @@
+    stop()
+    
+  }
+  
+  # if end year is not provided, use current year
+  if(rlang::is_empty(end_year)) {
+    
+    end_year <- Sys.Date() %>% substring(first = 1, last = 4)
+    
+  }
+  
+  # check whether season_start is provided when season_end is provided 
+  if(rlang::is_empty(season_start) & !rlang::is_empty(season_end)) {
+    
+    stop("Please always provide season_start if season_end is provided.")
+  }
+  
+  
+  # restrict time to season for which data should be retrieved
+  if(!rlang::is_empty(season_start)) {
+    
+    # account for seasons going across years
+    lst_periods <- purrr::map(.x = start_year:end_year,
+                              .f = ~{
+                                # create start and end date
+                                
+                                ## account for seasons going across years
+                                if(season_start[1] > season_end[1]) {
+                                  
+                                  start_date <- lubridate::make_date(.x - 1, season_start[1], season_start[2])
+                                  
+                                } else {
+                                  ## for seasons that start and end in the same year 
+                                  start_date <- lubridate::make_date(.x, season_start[1], season_start[2])
+                                  
+                                }
+                                
+                                end_date <- lubridate::make_date(.x, season_end[1], season_end[2])
+                                
+                                # create sequence of dates between start and end date
+                                start_seq <- seq(start_date, end_date - 1, by = 'days') %>% as.character()
+                                end_seq <- seq(start_date + 1, end_date, by = 'days') %>% as.character()
+                                
+                                return(tibble::lst(start_seq, end_seq))
+                                
+                              }
+    ) %>% 
+      purrr::list_c()
+    
+    
+    # bind list elements for each year together
+    start_date_seq <- lst_periods %>% purrr::keep(names(.) == "start_seq") %>% unlist()
+    end_date_seq <- lst_periods %>% purrr::keep(names(.) == "end_seq") %>% unlist()
+    
+  } else {
+    
+    # Create dates for the full year
+    
+    ## Create start and end date  
+    start_date <- lubridate::make_date(year = start_year, month = 1, day = 1)
+    end_date <- lubridate::make_date(year = end_year, month = 12, day = 31)
+    
+    ## Get sequence of dates between start and end date
+    start_date_seq <- seq(as.Date(start_date), as.Date(end_date) - 1, by = 'days') %>% as.character() 
+    end_date_seq <- seq(as.Date(start_date) + 1, as.Date(end_date), by = 'days') %>% as.character()
+    
+  }
+  
+  
+  # Create .nc file names per day
+  file_name <- list()
+  
+  file_name <- purrr::map(.x = 1:length(start_date_seq),
```

Comment:
```suggestion
  file_name <- purrr::map_chr(.x = 1:length(start_date_seq),
```
By adding the prefix `_chr`, the map function does not create a list but a vector of characters, so then you also don't have to rbind it later (at line176).